### ISSUE TITLE
Support for NPM, Yarn, and the greater Javascript ecosystem

### ISSUE THREAD

**Issue by vidjuheffex** [gambit/gambit#748]
As someone coming from the JS world. I think it would be great for Gambit to lean more into the practices and conventions of that ecosystem to increase adoption and exposure. 

For example, in the JS world people use [npm](https://www.npm.com) to distribute their code. This would result in installing Gambit-JS looking like:

`npm install gambit` or `yarn install gambit` depending on the user's choice of package manager.

In turn, using npm's `package.json`, users could be delivered what is needed by context.

In the browser, `import gambit` could pull in the browser-version, whereas in node, the same statement could pull in the node version. Lastly, `gambit` itself on the CLI could launch a node-based repl.

**Comment by feeley** [comment]
@vidjuheffex I agree that packaging Gambit so that it can be distributed through `npm` would be a nice feature.  My knowledge of JavaScript package managers is limited so maybe someone else is better placed to lead this effort.  I'm willing to help out once it is clear what packaging approach is the most appropriate.

@amirouche I don't fully understand your comment.  The file `VM.min.js` that is built in the `contrib/try` directory by a `make try` contains the Gambit interpreter and REPL.  This is all that is needed to run Scheme code in the browser or with `nodejs`.  In fact this is how the https://try.gambitscheme.org web app is built.

Note that `gsc` is typically bootstrapped using a C ecosystem because the Gambit repo contains files that have been precompiled from Scheme to C.  This provides high portability to Gambit because any platform with a C compiler can build Gambit, including the `gsc` compiler.  A `npm` packaging of Gambit could contain files that have been precompiled from Scheme to JavaScript, including `gsc`.

So the problem is not at the capability level but at the packaging level.
